network/p2p/middleware.go

```diff
@@ -392,17 +406,31 @@ func (m *Middleware) handleIncomingStream(s libp2pnetwork.Stream) {
 	}
 	_, isStaked := m.ov.Identities().ByNodeID(nodeID)
 
+	ctx, cancel := context.WithTimeout(m.ctx, LargeMsgUnicastTimeout)
+
+	deadline, _ := ctx.Deadline()
```

Comment:
do we not check the bool here since we already know there will be a deadline for sure?

Reply:
Yes, that's the reason.